test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,204 @@
+        double var24 = var23 + fmod(var23, 5);
+        double var25 = var24 + fmod(var24, 5);
+        double var26 = var25 + fmod(var25, 5);
+        double var27 = var26 + fmod(var26, 5);
+        double var28 = var27 + fmod(var27, 5);
+        double var29 = var28 + fmod(var28, 5);
+        double var30 = var29 + fmod(var29, 5);
+        double var31 = var30 + fmod(var30, 5);
+        double var32 = var31 + fmod(var31, 5);
+        double var33 = var32 + fmod(var32, 5);
+        double var34 = var33 + fmod(var33, 5);
+        double var35 = var34 + fmod(var34, 5);
+
+        double sum = var0 + var1 + var2 + var3 + var4 + var5 + var6 + var7 + var8 + var9 + var10 + var11 + var12
+                     + var13 + var14 + var15 + var16 + var17 + var18 + var19 + var20 + var21 + var22 + var23 + var24
+                     + var25 + var26 + var27 + var28 + var29 + var30 + var31 + var32 + var33 + var34 + var35;
+        printf("The sum is %5.2f, the argument is %lu ", sum, sizeOfArray);
+    }
+};
+
+using TestAccs = alpaka::test::EnabledAccs<alpaka::DimInt<1u>, std::uint32_t>;
+
+TEMPLATE_LIST_TEST_CASE("getFunctionAttributes.1D.withIdx", "[workDivKernel]", TestAccs)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Dim = alpaka::Dim<Acc>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    auto const& bundeledKernel = alpaka::KernelBundle(kernel, 200ul);
+
+    // Test getValidWorkDivForKernel for 1024 threads per grid
+    auto const workDiv = alpaka::getValidWorkDivForKernel<Acc>(dev, bundeledKernel, Vec{1024}, Vec{1});
+    // Test validity
+    auto const isValid = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv);
+    CHECK(isValid == true);
+
+    if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuCudaRt>)
+    {
+        // Get threads per block from the workDiv found by examining the kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent[0];
+        // Get hard limits
+        auto const kernelFunctionAttributes = alpaka::getFunctionAttributes<Acc>(bundeledKernel);
+        auto const threadsPerBlockLimit = kernelFunctionAttributes.maxThreadsPerBlock;
+        // Compare the workdiv value calculated using the kernel function to the device max limits for threads per
+        // block. The calculated value using kernel function is certainly less than or equal to the device max limits.
+        // For a common Nvidia GPU, expected values for threadsPerBlock and threadsPerBlockLimit are 512 and 1024.
+        // Depending on the GPU or the compiler the test below might fail because threadsPerBlock can be equal to
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock < static_cast<Idx>(threadsPerBlockLimit));
```

Comment:
I am wondering how this can become true? The function `getValidWorkDivForKernel` in line 90 to get the `workDiv` is using internally `kernelFunctionAttributes()` and in line 100 use call `kernelFunctionAttributes()` directly and compare `threadsPerBlock` with `threadsPerBlock`

Reply:
Thanks. I use `props.m_blockThreadCountMax` now.

Reply:
Shouldn't this be `<=` instead of `<` ?
Running with the same number of threads as `threadsPerBlockLimit` should be OK.